RESTResource/RESTResource.js

```diff
@@ -324,6 +324,17 @@ export default class RESTResource {
         options.params = options.params(parsedQuery, _.get(props, ['match', 'params']), mockProps(state, this.module, props.dataKey, this.logger).resources, this.logger, props);
       }
 
+      // headers
+      if (typeof options.headers === 'object') {
+        options.headers = _.mapValues(
+          options.headers,
+          header => substitute(header, props, state, this.module, this.logger, this.dataKey)
+        );
+      } else if (typeof options.headers === 'function') {
+        const parsedQuery = queryString.parse(_.get(props, ['location', 'search']));
+        options.headers = options.headers(parsedQuery, _.get(props, ['match', 'params']), mockProps(state, this.module, props.dataKey, this.logger).resources, this.logger, props);
```

Comment:
```suggestion
        const parsedQuery = queryString.parse(props?.location?.search);
        options.headers = options.headers(parsedQuery, props?.match?.params, mockProps(state, this.module, props.dataKey, this.logger).resources, this.logger, props);
```

Reply:
I think there are a lot of uses of `_.get`, predating the availability of the `?.` operator. I' not sure we necessarily want to fix these piecemeal.

Reply:
I'm all in favour of fixing things piecemeal. The fact is that nobody is gonna go, sit down, and remove all our reliance on lodash in a module in one swoop. So it's either we take our marginal gains here and there, or do nothing.

Reply:
As part of refactoring the building of the option object for `params` and `headers` into its own function, I took the opportunity to switch from `_.get` to optional chaining, it seemed a logic time to do it.

Reply:
Agree, given that this is now code that replaces the old inline version, it does make sense.